Style.md

```diff
@@ -0,0 +1,144 @@
+# Baseplate.go Coding Style Guide
+
+## Use tools
+
+Make sure your code passes the following tools:
+
+- `gofmt -s`
+- `golint`
+- `go vet`
+
+See our [IDE/Editor setup doc](Editor.md) for ways to doing that automatically.
+
+### Exceptions
+
+Generated code (e.g. code generated by thrift compiler) are exempted from this
+rule.
+
+## Import groups
+
+Group your imports into the following 3 groups, in that order, by blank lines:
+
+1. Standard library packages
+2. Third-party packages
+   (non-stdlib packages without `github.com/reddit/baseplate.go` prefix)
+3. Packages from the same module
+   (packages with `github.com/reddit/baseplate.go` prefix)
+
+Example:
+
+```go
+import (
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/log"
+)
+```
+
+### Exceptions
+
+In whole file examples
+(see the end of https://pkg.go.dev/testing?tab=doc#hdr-Examples),
+merge Groups 2&3.
+The reason is that whole file examples are presented in package users' point of
+view,
+and in that point of view everything under `github.com/reddit/baseplate.go` are
+also third-party packages.
+
+Please note that this exception does not include tests and normal examples.
+
+## Rename imports
+
+When an imported package name is different from the last element of the import
+path, always rename import it to make it explicit.
+
+Example:
+
+```go
+import (
+	jwt "gopkg.in/dgrijalva/jwt-go.v3"
+	opentracing "github.com/opentracing/opentracing-go"
+)
+```
+
+### Exceptions
+
+As recommended by go modules version design,
+in the example of `"github.com/go-redis/redis/v7"`,
+`v7` is the major version number,
+so we consider `redis` as the actual last element of import path.
+As a result there's no need to rename import it.
+
+## One-per-line style
+
+When putting all args to a function signature/call to a single line makes the
+line too long, Use one-per-line style with closing parenthesis on the next line.
```

Comment:
I think we need to clarify what too long here means.

Reply:
Go's official stance is that there's no limit on the length of line. Personally I use tabsize of 2 and try not to exceed 80, but that's not a hard limit (e.g. if a line just barely cross 80 I might let it slip). I think the thing we want to enforce is that "if you are not gonna put everything in the same line, the do it one-per-line instead". I'm open to suggestions on how to word that better.